components/org.wso2.carbon.identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java

```diff
@@ -203,63 +182,76 @@ public Response updateResource(String resourceId, ResourceDetailsDTO updatedReso
-                    log.error("Internal server error occurred. ", throwable);
+            if (isValidToken(context) && isResourceIdValid(resourceId)) {
+                if (ResourceUtils.getResourceService().deleteResource(resourceId)) {
+                    return Response.status(Response.Status.NO_CONTENT).build();
                 }
-            } else {
-                throw new UMAClientException(ResourceConstants.ErrorMessages.ERROR_CODE_NOT_FOUND_RESOURCE_ID);
             }
+        } catch (UMAServiceException e) {
+            log.error("Invalid request.", e);
+            handleErrorResponse(e, true);
+
         } catch (UMAClientException e) {
-            handleErrorResponse(e, log);
             log.error("Client error when retrieving resource from the resource server.", e);
-
+            handleErrorResponse(e, false);
         }
-        return response;
-
+        log.error("Resource can not be retrieved due to client side error.");
+        return Response.status(Response.Status.UNAUTHORIZED).entity(getErrorDTO
+                ("invalid request", "Error occurred in client side.")).build();
     }
 
     /**
      * Handle error responses.
      *
-     * @param umaException exception that should pass to handle error.
+     * @param throwable exception that should pass to handle error.
      */
-    public static void handleErrorResponse(UMAException umaException, Log log) throws ResourceEndpointException {
-
-        HandleErrorResponseConstants handleErrorResponseConstants = new HandleErrorResponseConstants();
-        if (handleErrorResponseConstants.getResponseMap().containsKey(umaException.getErrorCode())) {
-            String statusCode = handleErrorResponseConstants.getResponseMap().get(umaException.getErrorCode())[0];
-            Response.Status status = Response.Status.fromStatusCode(Integer.parseInt(statusCode));
-            String errorCode = handleErrorResponseConstants.getResponseMap().get(umaException.getErrorCode())[1];
-
-            throw buildResourceEndpointException(status, errorCode, umaException.getMessage());
+    private void handleErrorResponse(Throwable throwable, boolean isServerException)
+            throws ResourceEndpointException {
+
+        String code;
+        String errorCode = null;
+        Response.Status status = Response.Status.INTERNAL_SERVER_ERROR;
+        boolean isStatusOnly = true;
+        if (throwable instanceof UMAException) {
+            code = ((UMAException) throwable).getErrorCode();
+        } else {
+            code = ResourceConstants.ErrorMessages.ERROR_CODE_UNEXPECTED.getCode();
+        }
+        if (isServerException) {
+            if (throwable == null) {
+                log.error(status.getReasonPhrase());
+            } else {
+                log.error(status.getReasonPhrase(), throwable);
+            }
+        } else {
+            HandleErrorResponseConstants handleErrorResponseConstants = new HandleErrorResponseConstants();
```

Comment:
Consent inside this is public and static right ? Do we need to instantiate this ?